### PR TITLE
Document large_allocation_warning_threshold node property

### PR DESCRIPTION
**Pull request by shorgi**
Fixes:
- #1176 

Preview: 
- Added [new node property](https://deploy-preview-1179--redpanda-documentation.netlify.app/docs/beta/reference/node-properties/#large_allocation_warning_threshold)

~~Blocked by~~ Depends on https://github.com/redpanda-data/redpanda/pull/8596

### CODE REVIEW

**Comment by shorgi** [comment]
@ballard26 - this docs PR is ready for your review, but it won't be merged until https://github.com/redpanda-data/redpanda/pull/8596 is completed.

**Comment by Feediver1** [comment]
@shorgi https://github.com/redpanda-data/redpanda/pull/8596 was merged. Is this still blocked? No updates here since that dependency/blocker was merged. Is there another blocker?
@ballard26 If this PR can be merged, please approve. If not, can you let us know what needs to be updated? Thx.

**Comment by shorgi** [comment]
> @shorgi [redpanda-data/redpanda#8596](https://github.com/redpanda-data/redpanda/pull/8596) was merged. Is this still blocked? No updates here since that dependency/blocker was merged. Is there another blocker? 

@Feediver1 no additional blocker that I'm aware of, removing that from this issue, once @ballard26 approves this will be merged

File: docs/reference/node-properties.mdx [hunk 1]
```diff
@@ -135,6 +135,20 @@ Transport Layer Security (TLS) configuration for the Kafka API endpoint.
 
 ---
 
+### large_allocation_warning_threshold
+
+Enables log warning messages for memory allocations greater than or equal to the given threshold size, in bytes.
```
Comment (ballard26) [inline docs/reference/node-properties.mdx:140]: This should be just `greater than` rather than `greater than or equal to`

**Pull request merged**